### PR TITLE
Add Codelab Sample App

### PR DESCRIPTION
**Pull request by meltsufin**

### CODE REVIEW

File: google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/README.adoc [hunk 1]
```diff
@@ -0,0 +1,5 @@
+= Cloud Spanner Hibernate Codelab
+
+This is the solution for the Cloud Spanner with Hibernate ORM Codelab.
+
+Please refer to the codelab for instruction
```
Comment (dzou) [inline google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/README.adoc]: Add period at end of sentence.

File: google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.Date;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.MetadataSources;
+import org.hibernate.boot.registry.StandardServiceRegistry;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+
+/**
+ * The main app.
```
Comment (dzou) [inline google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java]:
```suggestion
 * The Hibernate codelab solution which demonstrates basic operations in Hibernate such as reading, writing, and deleting data.
```

File: google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+
+    writeData(session);
+
+    readData(session);
+
+    // close Hibernate session and sessionFactory
+    session.close();
+    sessionFactory.close();
+  }
+
```
Comment (dzou) [inline google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java]: Would be useful to add a javadoc for each private method to describe what each is doing. I.e. readData is not simply doing a basic `Select *`

File: google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/README.adoc (hunk 1 above)
Comment (dzou) [reply google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/README.adoc]: Also I would suggest circling back here and adding a link from the docs to the codelab once it is ready. You can create an issue as a reminder to do this.

File: google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+  private static void clearData(Session session) {
+    session.beginTransaction();
+
+    session.createQuery("delete from Album where 1=1").executeUpdate();
```
Comment (dmitry-s) [inline google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java]: Maybe change to `where true`?

File: google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+
+  private static void writeData(Session session) {
+    session.beginTransaction();
+
+    Singer singerMelissa = new Singer("Melissa", "Garcia", makeDate("1981-03-19"));
+    Album albumGoGoGo = new Album(singerMelissa, "Go, Go, Go");
+    session.save(singerMelissa);
+    session.save(albumGoGoGo);
+
+    session.save(new Singer("Russell", "Morales", makeDate("1978-12-02")));
```
Comment (dmitry-s) [inline google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java]: moving `makeDate` call into the constructor would reduce duplication

Comment (meltsufin) [reply google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java]: good point

Comment (meltsufin) [reply google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/main/java/codelab/App.java]: The only problem is that it couldn't be a private method anymore.

File: google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/test/java/codelab/AppTests.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+  @Rule
+  public final SystemOutRule systemErrRule = new SystemOutRule().enableLog();
+
+  @Test
+  public void test() {
+    App.main(new String[]{});
+
+    assertThat(systemErrRule.getLog()).contains("Singers who were born in 1990 or later:\n"
+        + "Jacqueline Long born on 1990-07-29\n"
```
Comment (ChengyuanZhao) [inline google-cloud-spanner-hibernate-samples/spanner-hibernate-codelab/src/test/java/codelab/AppTests.java]: nice

**Pull request merged**